src/components/UploadPage.jsx

```diff
@@ -31,14 +31,30 @@ const UploadPage = () => {
                 <input style={textInputStyle} type="email" value={email} onChange={handleChange} placeholder={"Email"} />
                 <UploadBox setFiles={setFiles} />
                 <div>{files.map(f => f.path)}</div>
-                <input type="submit" value="Submit" style={{...itemStyle, ...textInputStyle}} />
+                <input className="g-recaptcha" data-sitekey="6LeIkakaAAAAAN7IyMjgyCZphTmLkgQuPC0uEh5K" data-callback="handleCaptcha" type="submit" value="Submit" style={{ ...itemStyle, ...textInputStyle }} />
             </form>
         </div>
         <h1>{displayText}</h1>
         </>
     )
 }
 
+// Added global function to fetch CAPTCHA response data (DOES NOT WORK LOCALHOST DUE TO CORS. UNTESTED FOR HEROKU)
+// Ideal scenario is that it will produce in console all the CAPTCHA data, untested because of CORS
+function handleCaptcha(event) {
+    fetch("https://www.google.com/recaptcha/api/siteverify", {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
+            secret: '6LeIkakaAAAAAEVtMRSeU2Zhoba4x5P2E_bPvg8r',
```

Comment:
Pretty sure this isn't allowed to be in the repo
https://developers.google.com/recaptcha/intro
"The secret key needs to be kept safe for security purposes."
The usual way to include secrets to have a secrets.json file in the repo, which is not committed, and we share that as needed.
Note that you can't just revert the commit, since it will be in the git history still and everyone will be able to see it.
You'll need to do something like: 
`git reset HEAD^ --soft` to remove the commit locally (while not losing the changes), then move this into a secrets file, commit the changes and force push `git push -f`

Reply:
Scratch that, that isn't going to work since it's the people using the app need the key

Reply:
Okay, I understand now I think. This is meant to happen on the server, so they secret can remain secret.

Reply:
Rather than a secrets.json file, it would be easier to deploy if you made it an environment variable